Assets/SEE/GraphProviders/VCSGraphProvider.cs

```diff
@@ -0,0 +1,387 @@
+        /// If not, exceptions are thrown accordingly.
+        /// </summary>
+        /// <param name="city">To be checked</param>
+        /// <exception cref="ArgumentException">thrown in case <see cref="RepositoryPath"/>,
+        /// or <see cref="CommitID"/>
+        /// is undefined or does not exist or <paramref name="city"/> is null</exception>
+        protected void CheckArguments(AbstractSEECity city)
+        {
+            if (string.IsNullOrEmpty(RepositoryPath.Path))
+            {
+                throw new ArgumentException("Empty repository path.\n");
+            }
+            if (!Directory.Exists(RepositoryPath.Path))
+            {
+                throw new ArgumentException($"Directory {RepositoryPath.Path} does not exist.\n");
+            }
+            if (string.IsNullOrEmpty(CommitID))
+            {
+                throw new ArgumentException("Empty CommitID.\n");
+            }
+            if (city == null)
+            {
+                throw new ArgumentException("The given city is null.\n");
+            }
+        }
+
+        /// <summary>
+        /// The node type for directories.
+        /// </summary>
+        private const string directoryNodeType = "Directory";
+        /// <summary>
+        /// The node type for files.
+        /// </summary>
+        private const string fileNodeType = "File";
+
+        /// <summary>
+        /// Builds the VCS graph with specific metrics.
+        /// </summary>
+        /// <param name="pathGlobbing">The paths which get included/excluded.</param>
+        /// <param name="repositoryPath">The path to the repository.</param>
+        /// <param name="commitID">The commitID where the files exist.</param>
+        /// <returns>the graph.</returns>
+        private static Graph GetVCSGraph(Dictionary<string, bool> pathGlobbing, string repositoryPath, string commitID)
+        {
+            string[] pathSegments = repositoryPath.Split(Path.DirectorySeparatorChar);
+
+            Graph graph = new(repositoryPath, pathSegments[^1]);
+            graph.CommitID(commitID);
+            graph.RepositoryPath(repositoryPath);
+
+            // The main directory.
+            NewNode(graph, pathSegments[^1], directoryNodeType, pathSegments[^1]);
+
+            IEnumerable<string> includedPathGlobs = pathGlobbing
+                .Where(path => path.Value)
+                .Select(path => path.Key).ToHashSet();
+
+            IEnumerable<string> excludedPathGlobs = pathGlobbing
+                .Where(path => !path.Value)
+                .Select(path => path.Key).ToHashSet();
+
+            using (Repository repo = new(repositoryPath))
+            {
+                LibGit2Sharp.Tree tree = repo.Lookup<Commit>(commitID).Tree;
+                // Get all files using "git ls-tree -r <CommitID> --name-only".
+                IEnumerable<string> files;
+                if (includedPathGlobs.Any() && !string.IsNullOrEmpty(includedPathGlobs.First()))
```

Comment:
This is not standard path globbing. Why aren't you just using the [standard  API of .NET](https://learn.microsoft.com/en-us/dotnet/core/extensions/file-globbing)?

Moreover, mind that the order of globbings as given by the user must be considered. Your code does not do that.